lib/upgrade/etcd/conversionv1v3/node.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+type Node struct{}
+
+// convertKVPairToAPI converts a KVPair containing a backend Node and NodeKey
+// to an API Node structure.
+// This is part of the conversionHelper interface.
+func (_ Node) BackendV1ToAPIV3(d *model.KVPair) (Resource, error) {
+	bv, ok := d.Value.(*model.Node)
+	if !ok {
+		return nil, fmt.Errorf("Value is not a valid Node resource: %v", d.Value)
+	}
+
+	bk, ok := d.Key.(model.NodeKey)
+	if !ok {
+		return nil, fmt.Errorf("Key is not a valid NodeKey resource: %v", d.Value)
+	}
+
+	apiNode := apiv3.NewNode()
+
+	apiNode.ObjectMeta.Name = convertName(bk.Hostname)
```

Comment:
We'll need to use the `convertNodeName` function when it exists.

Reply:
See if you like what I did with the alias function.